### DISCUSSION TITLE
More about editing files

### DISCUSSION THREAD

**Discussion by AASabancev** [electerm/electerm#3571] [Ideas]
1. no SSH client provides the ability to go to sudo su to edit the necessary files. It would be ideal to go to the terminal and then go to the necessary folders and edit the necessary files.

2. in the current version 1.39.109-mac-arm64 something strange happens - when saving a file, the editor window seems to save - but does not save, there are no errors, the file does not change. It seems there are no rights to the file. but there are no errors either.

3. when editing a file in a third-party editor - saving also does not work even for the file /tmp/test.txt , nothing just happens

4. a progress bar for uploaded files is needed, if they ended with an error - report this with the ability to try again.

**Comment by zxdong262** [comment]
Thank you for the feedback, would fix in future releases

**Comment by AASabancev** [comment]
Hello! 
just updated version to electerm-1.40.6-mac-arm64 an checks the bugs above:
2 - doesn't fixed
3 - doesn't fixed. In addition to point 3: it is still not clear - when changing a file in an external editor - is the changed file uploaded to FTP or not, it is better to ask: that the file has changed - upload?

5 new bug: if the external editor is not defined (MacOS) then there will be an error "No application knows how to open" and file editing will hang
6. new proposal: we need the ability to not use the built-in file editor, so that the external one opens immediately. and the application will track the file change and offer to upload the changed file to FTP.

thank you

**Reply by zxdong262** [reply]
Created a issue to track this: #3604
